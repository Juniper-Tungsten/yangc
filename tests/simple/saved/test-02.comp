<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:yin="urn:ietf:params:xml:ns:yang:yin:1" xmlns:yangc="http://juise.org/yangc/1.0" version="1.0">
  <xsl:param name="MIN" select="12"/>
  <xsl:param name="MAX" select="100"/>
  <xsl:param name="test-default" select="42"/>
  <xsl:param name="test" select="&quot;not(foo)&quot;"/>
  <xsl:param name="this" select="&quot;one&quot;"/>
  <xsl:param name="that" select="&quot;two&quot;"/>
  <xsl:template name="foo">
    <xsl:param name="a"/>
    <xsl:param name="b"/>
    <xsl:param name="c"/>
    <yin:leaf name="foo-is-working">
      <xsl:if test="$c &gt; 1">
        <yin:description>
          <yin:text>
            <xsl:value-of select="$b"/>
          </yin:text>
        </yin:description>
      </xsl:if>
    </yin:leaf>
  </xsl:template>
  <xsl:template match="/features">
    <yin:module xmlns="http://foo.example.net" xmlns:foo="http://foo.example.net" name="foo">
      <!-- This is a commen -->
      <!-- This is a comment also -->
      <yin:namespace uri="http://foo.example.net">
        <yin:prefix value="foo"/>
      </yin:namespace>
      <yin:contact>
        <yin:text>
          <xsl:text>Phil Shafer</xsl:text>
        </yin:text>
      </yin:contact>
      <yin:include module="base-02"/>
      <yin:augment target-node="protocols">
        <xsl:call-template name="foo">
          <xsl:with-param name="a" select="&quot;foo-data&quot;"/>
          <xsl:with-param name="b" select="&quot;Foo Data&quot;"/>
          <xsl:with-param name="c" select="2"/>
        </xsl:call-template>
        <yin:container name="one">
          <yin:must condition="not(../two == {$MAX})"/>
          <yin:must condition="{$test}"/>
          <yin:must condition="{$this} == {$that}"/>
          <yin:description>
            <yin:text>
              <xsl:text>This is a help string</xsl:text>
            </yin:text>
          </yin:description>
          <yin:config value="true"/>
        </yin:container>
        <yin:leaf name="test">
          <yin:description>
            <yin:text>
              <xsl:text>testing</xsl:text>
            </yin:text>
          </yin:description>
          <yin:must condition="test + ing"/>
          <yin:type name="int">
            <yin:range value="{$MIN} .. {$MAX}"/>
          </yin:type>
          <yin:default value="{$test-default}"/>
          <yin:must condition=". &lt; {$MAX}"/>
        </yin:leaf>
      </yin:augment>
      <yin:container name="examples-from-spec">
        <yin:leaf name="host-name">
          <yin:type name="string"/>
          <yin:description>
            <yin:text>
              <xsl:text>Hostname for this system</xsl:text>
            </yin:text>
          </yin:description>
        </yin:leaf>
        <yin:container name="system">
          <yin:container name="login">
            <yin:leaf name="message">
              <yin:type name="string"/>
              <yin:description>
                <yin:text>
                  <xsl:text>Message given at start of login session</xsl:text>
                </yin:text>
              </yin:description>
            </yin:leaf>
            <yin:list name="user">
              <yin:key value="name"/>
              <yin:leaf name="name">
                <yin:type name="string"/>
              </yin:leaf>
              <yin:leaf name="full-name">
                <yin:type name="string"/>
              </yin:leaf>
              <yin:leaf name="class">
                <yin:type name="string"/>
              </yin:leaf>
            </yin:list>
          </yin:container>
        </yin:container>
      </yin:container>
    </yin:module>
  </xsl:template>
</xsl:stylesheet>
