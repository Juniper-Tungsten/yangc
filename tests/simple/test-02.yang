module foo {
    // This is a comment
    /* This is a comment also */
    namespace "http://foo.example.net" {
        prefix foo;
    }
    contact "Phil Shafer";

    param $MIN = 12;
    param $MAX = 100;
    param $test-default = 42;
    param $test = "not(foo)";
    param $this = "one";
    param $that = "two";

    include base-02;

    template foo ($a, $b, $c) {
    	leaf foo-is-working {
            if ($c > 1) {
                help $b;
            }
        }
    }

    augment protocols {
        call foo($a = "foo-data", $b = "Foo Data", $c = 2);

	container one {
	    must not(../two == $MAX);
	    must $test;
	    must $this==$that;
	    help "This is" + " a help string";
	    config true;
	}

        leaf test {
            help "test" + "ing";
	    must "test" + "ing";
            type int {
                range $MIN .. $MAX;
            }
            default $test-default;
            must . < $MAX;
        }
    }

    container examples-from-spec {
	leaf host-name {
	    type string;
	    description "Hostname for this system";
	}

	
	container system {
	    container login {
		leaf message {
		    type string;
		    description 
			"Message given at start of login session";
		}

		list user {
		    key "name";
		    leaf name {
			type string;
		    }
		    leaf full-name {
			type string;
		    }
		    leaf class {
			type string;
		    }
		}
	    }
	}
    }
}
