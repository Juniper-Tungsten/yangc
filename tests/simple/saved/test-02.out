<?xml version="1.0"?>
<yin:module xmlns="http://foo.example.net" xmlns:foo="http://foo.example.net" xmlns:yin="urn:ietf:params:xml:ns:yang:yin:1" xmlns:yangc="http://juise.org/yangc/1.0" name="foo">
  <yin:namespace uri="http://foo.example.net">
    <yin:prefix value="foo"/>
  </yin:namespace>
  <yin:contact>
    <yin:text>Phil Shafer</yin:text>
  </yin:contact>
  <yin:include module="base-02"/>
  <yin:augment target-node="protocols">
    <yin:leaf name="foo-is-working">
      <yin:description>
        <yin:text>Foo Data</yin:text>
      </yin:description>
    </yin:leaf>
    <yin:container name="one">
      <yin:must condition="not(../two == 100)"/>
      <yin:must condition="not(foo)"/>
      <yin:must condition="one == two"/>
      <yin:description>
        <yin:text>This is a help string</yin:text>
      </yin:description>
      <yin:config value="true"/>
    </yin:container>
    <yin:leaf name="test">
      <yin:description>
        <yin:text>testing</yin:text>
      </yin:description>
      <yin:must condition="test + ing"/>
      <yin:type name="int">
        <yin:range value="12 .. 100"/>
      </yin:type>
      <yin:default value="42"/>
      <yin:must condition=". &lt; 100"/>
    </yin:leaf>
  </yin:augment>
  <yin:container name="examples-from-spec">
    <yin:leaf name="host-name">
      <yin:type name="string"/>
      <yin:description>
        <yin:text>Hostname for this system</yin:text>
      </yin:description>
    </yin:leaf>
    <yin:container name="system">
      <yin:container name="login">
        <yin:leaf name="message">
          <yin:type name="string"/>
          <yin:description>
            <yin:text>Message given at start of login session</yin:text>
          </yin:description>
        </yin:leaf>
        <yin:list name="user">
          <yin:key value="name"/>
          <yin:leaf name="name">
            <yin:type name="string"/>
          </yin:leaf>
          <yin:leaf name="full-name">
            <yin:type name="string"/>
          </yin:leaf>
          <yin:leaf name="class">
            <yin:type name="string"/>
          </yin:leaf>
        </yin:list>
      </yin:container>
    </yin:container>
  </yin:container>
</yin:module>
module foo {
    namespace "http://foo.example.net" {
        prefix foo;
    }
    contact "Phil Shafer";
    include base-02;
    augment protocols {
        leaf foo-is-working {
            description "Foo Data";
        }
        container one {
            must "not(../two == 100)";
            must not(foo);
            must "one == two";
            description "This is a help string";
            config true;
        }
        leaf test {
            description testing;
            must "test + ing";
            type int {
                range "12 .. 100";
            }
            default 42;
            must ". < 100";
        }
    }
    container examples-from-spec {
        leaf host-name {
            type string;
            description "Hostname for this system";
        }
        container system {
            container login {
                leaf message {
                    type string;
                    description "Message given at start of login session";
                }
                list user {
                    key name;
                    leaf name {
                        type string;
                    }
                    leaf full-name {
                        type string;
                    }
                    leaf class {
                        type string;
                    }
                }
            }
        }
    }
}

